File: Logica_de_Programacao_Python/Desafio_8/desafio_8/Desafio_8.py
while True:
    numero = int(input("Digite um número entre 1 e 10: "))

    if 1 <= numero <= 10:
        break
    else:
        print("Valor inválido! Digite um número entre 1 e 10.\n")

print(f"\nTabuada do {numero}:")
for i in range(1, 11):
    resultado = numero * i
    print(f"{numero} x {i} = {resultado}")

File: Logica_De_Programacao_Java/Desafio_8/src/Desafio_8.java
import java.util.Scanner;

public class Desafio_8 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numero;

        do {
            System.out.print("Digite um número inteiro entre 1 e 10: ");
            numero = scanner.nextInt();

            if (numero < 1 || numero > 10) {
                System.out.println("Número inválido! Por favor, insira um número entre 1 e 10.");
            }
        } while (numero < 1 || numero > 10);

        System.out.println("\nTabuada do " + numero + ":");
        for (int i = 1; i <= 10; i++) {
            System.out.println(numero + " x " + i + " = " + (numero * i));
        }
        scanner.close();
    }
}